packages/eip-5792-middleware/src/utils.ts

```diff
@@ -0,0 +1,38 @@
+import type { KeyringTypes } from '@metamask/keyring-controller';
+import { JsonRpcError } from '@metamask/rpc-errors';
+import type { Hex } from '@metamask/utils';
+
+import { EIP5792ErrorCode } from './constants';
+import type { EIP5792Messenger } from './types';
+
+/**
+ * Retrieves the keyring type for a given account address.
+ *
+ * @param accountAddress - The account address to look up.
+ * @param messenger - Messenger instance for controller communication.
+ * @returns The keyring type associated with the account.
+ * @throws JsonRpcError if the account type is unknown or not found.
+ */
+export function getAccountKeyringType(
```

Comment:
Minor, is there value in writing unit tests for this so we can mock it in other test files?